src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/model/BuildMemory.java

```diff
@@ -406,6 +408,24 @@ public void setEntryFailureMessage(GerritTriggeredEvent event, Run r, String fai
     }
 
     /**
+     * Creates a snapshot clone of the current coordination memory status.
+     *
+     * @return the report
+     */
+    @Nonnull
+    public synchronized BuildMemoryReport report() {
+        BuildMemoryReport report = new BuildMemoryReport();
+        for (Map.Entry<GerritTriggeredEvent, MemoryImprint> entry : memory.entrySet()) {
+            List<Entry> triggered = new LinkedList<Entry>();
```

Comment:
An ArrayList will have better cache locality when iterating and the many nodes of a LinkedList make it worse in terms of wasted memory.

Unlearning needs of you, young padawan.

Use a LinkedList when many inserts you expect. For append-only ArrayList is much better than you think

Sent from my iPhone

> On 10 Mar 2016, at 11:47, Robert Sandell <notifications@github.com> wrote:
> 
> In src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/model/BuildMemory.java:
> 
> > @@ -406,6 +408,24 @@ public void setEntryFailureMessage(GerritTriggeredEvent event, Run r, String fai
> >      }
> >  
> >      /**
> > +     * Creates a snapshot clone of the current coordination memory status.
> > +     *
> > +     * @return the report
> > +     */
> > +    @Nonnull
> > +    public synchronized BuildMemoryReport report() {
> > +        BuildMemoryReport report = new BuildMemoryReport();
> > +        for (Map.Entry<GerritTriggeredEvent, MemoryImprint> entry : memory.entrySet()) {
> > +            List<Entry> triggered = new LinkedList<Entry>();
> Because an ArrayList preallocates memory that might not be needed, and when that memory is full it needs to allocate even more new memory and move that over.
> According to what basic algorithm theory taught me in school; an ArrayList is only effective to use when you need to access specific indices in the list. Everywhere else it is ineffective.
> Since this list will only be used to iterate over from start to end once or twice, a LinkedList is more effective to use.
> 
> —
> Reply to this email directly or view it on GitHub.
> 
> -- 
> You received this message because you are subscribed to the Google Groups "engineering-code-reviews" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to engineering-code-reviews+unsubscribe@cloudbees.com.
> To post to this group, send email to engineering-code-reviews@cloudbees.com.
> To view this discussion on the web visit https://groups.google.com/a/cloudbees.com/d/msgid/engineering-code-reviews/jenkinsci/gerrit-trigger-plugin/pull/275/r55668111%40github.com.